tools/Customization/DevHome.Customization/ViewModels/DevDriveInsightsViewModel.cs

```diff
@@ -0,0 +1,302 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using DevHome.Common.Models;
+using DevHome.Common.Services;
+using DevHome.Customization.ViewModels.Environments;
+using DevHome.Customization.Views;
+using DevHome.SetupFlow.Services;
+using DevHome.SetupFlow.ViewModels;
+using Serilog;
+
+namespace DevHome.Customization.ViewModels;
+
+public partial class DevDriveInsightsViewModel : ObservableObject
+{
+    private readonly Microsoft.UI.Dispatching.DispatcherQueue _dispatcher;
+
+    public ObservableCollection<DevDriveCardViewModel> DevDriveCardCollection { get; private set; } = new();
+
+    public ObservableCollection<DevDriveOptimizerCardViewModel> DevDriveOptimizerCardCollection { get; private set; } = new();
+
+    public ObservableCollection<DevDriveOptimizedCardViewModel> DevDriveOptimizedCardCollection { get; private set; } = new();
+
+    private readonly OptimizeDevDriveDialogViewModelFactory _optimizeDevDriveDialogViewModelFactory;
+
+    [ObservableProperty]
+    private bool _shouldShowCollectionView;
+
+    [ObservableProperty]
+    private bool _shouldShowOptimizerCollectionView;
+
+    [ObservableProperty]
+    private bool _shouldShowOptimizedCollectionView;
+
+    [ObservableProperty]
+    private bool _devDriveLoadingCompleted;
+
+    [ObservableProperty]
+    private bool _devDriveOptimizerLoadingCompleted;
+
+    [ObservableProperty]
+    private bool _devDriveOptimizedLoadingCompleted;
+
+    public IDevDriveManager DevDriveManagerObj { get; private set; }
+
+    private IEnumerable<IDevDrive> ExistingDevDrives { get; set; } = Enumerable.Empty<IDevDrive>();
+
+    public DevDriveInsightsViewModel(IDevDriveManager devDriveManager, OptimizeDevDriveDialogViewModelFactory optimizeDevDriveDialogViewModelFactory)
+    {
+        _optimizeDevDriveDialogViewModelFactory = optimizeDevDriveDialogViewModelFactory;
+        _dispatcher = Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread();

```

Comment:
It isn't guaranteed that this will run on the UI thread every time. I can forward you the internal thread I started about this. This can return null and will result in a crash.